### ISSUE TITLE
JUCE is pulled twice when cloning recursively

### ISSUE THREAD

**Issue by alex-tee** [utokusa/OS-251#16]
```
Submodule 'lib/JUCE' (https://github.com/juce-framework/JUCE) registered for path 'lib/JUCE'
Submodule 'lib/react-juce' (https://github.com/nick-thompson/react-juce) registered for path 'lib/react-juce'
******* Cloning into '/home/alex/Documents/aur/os-251/src/os-251/lib/JUCE'...
Cloning into '/home/alex/Documents/aur/os-251/src/os-251/lib/react-juce'...
Submodule path 'lib/JUCE': checked out '7c797c8105c2d41872e6e8d08972624f0afd335d'
Submodule path 'lib/react-juce': checked out '2711facc9c064daffeab4f331cb472fa0985585f'
Submodule 'juce' (https://github.com/juce-framework/JUCE.git) registered for path 'lib/react-juce/ext/juce'
Submodule 'yoga' (https://github.com/facebook/yoga.git) registered for path 'lib/react-juce/react_juce/yoga'
******* Cloning into '/home/alex/Documents/aur/os-251/src/os-251/lib/react-juce/ext/juce'...
```

**Comment by utokusa** [comment]
The cause of the problem is that OS-251 has two git submodules. One is JUCE (`lib/JUCE`) and the other is  react-juce (`lib/react-juce`) which also has JUCE as its submodule.
I guess I can resolve it by following https://stackoverflow.com/questions/4115817/duplicate-submodules-with-git .

But currently I don''t want to use the solution  because:
- I want keep the repository structure simple.
- The current structure gives me  some flexibility. 
   e.g. 
  1. Upgrading lib/JUCE while leaving lib/react-juce/JUCE as it is. It helped me to adress #15 without modifying react-juce.
  2. Use https://github.com/lv2-porting-project/JUCE for lib/JUCE to provide LV2 version without modifying react-juce.

What do you think?

**Comment by alex-tee** [comment]
I see currently both JUCE versions are the same commit (`7c797c8105c2d41872e6e8d08972624f0afd335d`) so that's fine for now, but if you decide to switch the main JUCE module (for example to https://github.com/lv2-porting-project/JUCE) I think it's best to make sure that react-juce also uses the same JUCE module to avoid trouble in the future. I'm not exactly sure how react-juce works but maybe the 2 JUCE versions will be incompatible or have runtime differences that will cause bugs.

You could use something like a "dist" script to prepare release tarballs (just copy/symlink your main JUCE directory to where react-juce expects JUCE to be when preparing a release tarball).

**Comment by utokusa** [comment]
> 2 JUCE versions will be incompatible or have runtime differences that will cause bugs.

Agree.  I have to choose between 1. Same JUCE but lib/react-juce/ext/JUCE is updated by me to fix bug #15 2. Use 2 JUCE version. But currently I'm not sure which is better.

> You could use something like a "dist" script to prepare release tarballs (just copy/symlink your main JUCE directory to where react-juce expects JUCE to be when preparing a release tarball).

Thanks. So using symlink looks like the ideal option.
